lib/puppet/type/cert_key_bundle.rb

```diff
@@ -0,0 +1,96 @@
+Puppet::Type.newtype(:cert_key_bundle) do
+  desc 'combines a certificate and private key into a single file'
+
+  ensurable
+
+  newparam(:path, :namevar => true) do
+    desc "Path to the file that will contain the certificate and private key bundle"
+    isrequired
+  end
+
+  newparam(:certificate) do
+    desc "Path to certificate file to include in bundle"
+    isrequired
+
+    def insync?(is)
+      File.read(self[:path]).include?(File.read(should))
+    end
+  end
+
+  newparam(:private_key) do
+    desc "Path to private key file to include in bundle"
+    isrequired
+
+    def insync?(is)
+      should = format_as_pkcs_1(should) if self[:force_pkcs_1]
+      File.read(self[:path]).include?(File.read(should))
+    end
+  end
+
+  newparam(:owner, parent: Puppet::Type::File::Owner) do
+    desc "Specifies the owner of the file. Valid options: a string containing a username or integer containing a uid."
+  end
+
+  newparam(:group, parent: Puppet::Type::File::Group) do
+    desc "Specifies a permissions group for the file. Valid options: a string containing a group name or integer containing a gid."
+  end
+
+  newparam(:mode, parent: Puppet::Type::File::Mode) do
+    desc "Specifies the permissions mode of the file. Valid options: a string containing a permission mode value in octal notation."
+  end
+
+  newparam(:force_pkcs_1, :boolean => true, :parent => Puppet::Parameter::Boolean) do
+    desc "Ensures the private key is in PKCS#1 format. This is required for services like Apache reverse proxy"
+  end
+
+  def create
+    File.write(self[:path], bundle)
+  end
+
+  def exists?
+    return true if File.exist?(self[:path])
+    bundle == File.read(self[:path])
+  end
+
+  def destroy
+    File.rm(self[:path])
+  end
+
+  def bundle
+    cert = File.read(self[:certificate])
+    key = File.read(self[:private_key])
+
+    key = format_as_pkcs_1(key) if self[:force_pkcs_1]
+
+    [key, cert].join("\n")
+  end
+
+  def format_as_pkcs_1(content)
+    content.gsub(/(BEGIN|END) (PRIVATE KEY)/, '\1 RSA \2')
+  end
```

Comment:
```suggestion
  def format_as_pkcs_1(content)
    OpenSSL::PKey::RSA.new(content).to_pem
  end
```

I think this works better than my previous suggestion. It uses a purpose built tool without requiring a separate provider, or any hacking to deal with Puppet's stderr/stdout weirdness.

The price is adding `require 'openssl'` to the type and adding the dependency to the Gemfile.

Reply:
> The price is adding `require 'openssl'` to the type and adding the dependency to the Gemfile.

AFAIK `openssl` is in Ruby itself (https://ruby-doc.org/stdlib-2.7.1/libdoc/openssl/rdoc/OpenSSL.html) so there shouldn't be a need to add it to the Gemfile. And Puppet itself probably already uses is so while I'd still add `require 'openssl'`, it's probably a noop.